src/components/Modal/index.tsx

```diff
@@ -1,69 +1,262 @@
       onClose={handleClose}
       open={open}
     >
-      <div className={cn(classes.paper, paperClassName, 'classpep')}>{children}</div>
-    </Modal>
+      <div className={cn('paper', paperClassName)}>{children}</div>
+    </ModalStyled>
   )
 }
 
 export default GnoModal
+
+/*****************/
+/* Generic Modal */
+/*****************/
+
+/*** Header ***/
+const HeaderSection = styled.div`
+  display: flex;
+  padding: 24px 18px 24px 24px;
+  border-bottom: 2px solid ${({ theme }) => theme.colors.separator};
+
+  h5 {
+    color: ${({ theme }) => theme.colors.text};
+  }
+
+  .close-button {
+    align-self: flex-end;
+    background: none;
+    border: none;
+    padding: 5px;
+    width: 26px;
+    height: 26px;
+
+    span {
+      margin-right: 0;
+    }
+
+    :hover {
+      background: ${({ theme }) => theme.colors.separator};
+      border-radius: 16px;
+      cursor: pointer;
+    }
+  }
+`
+
+const TitleStyled = styled(TitleSRC)`
+  display: flex;
+  align-items: center;
+  flex-basis: 100%;
+
+  .image,
+  img {
+    width: 20px;
+    margin-right: 10px;
+  }
+
+  .note,
+  span {
+    margin-left: 12px;
+  }
+`
+
+interface TitleProps {
+  children: string | ReactNode
+  size?: keyof Theme['title']['size']
+  withoutMargin?: boolean
+  strong?: boolean
+}
+
+const Title = ({ children, ...props }: TitleProps): ReactElement => (
+  <TitleStyled size="xs" withoutMargin {...props}>
+    {children}
+  </TitleStyled>
+)
+
+interface HeaderProps {
+  children?: ReactNode
+  onClose?: (event: any) => void
+}
+
+const Header = ({ children, onClose }: HeaderProps): ReactElement => {
+  return (
+    <HeaderSection className="modal-header">
+      {children}
+
+      {onClose && (
+        <button className="close-button" onClick={onClose}>
+          <Icon size="sm" type="cross" />
+        </button>
+      )}
+    </HeaderSection>
+  )
+}
+
+Header.Title = Title
+
+/*** Body ***/
+const BodySection = styled.div<{ withoutPadding: BodyProps['withoutPadding'] }>`
+  padding: ${({ withoutPadding }) => (withoutPadding ? 0 : '24px')};
+`
+
+interface BodyProps {
+  children: ReactNode | ReactNodeArray
+  withoutPadding?: boolean
+}
+
+const Body = ({ children, withoutPadding = false }: BodyProps): ReactElement => (
+  <BodySection className="modal-body" withoutPadding={withoutPadding}>
+    {children}
+  </BodySection>
+)
+
+/*** Footer ***/
+const FooterSection = styled.div`
+  display: flex;
+  justify-content: center;
+  border-top: 2px solid ${({ theme }) => theme.colors.separator};
+  padding: 24px;
+`
+
+const ButtonStyled = styled(Button)`
+  &.MuiButtonBase-root {
+    margin: 0 10px;
+  }
+`
+
+interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
+  text?: string
+  size?: keyof Theme['buttons']['size']
+  color?: 'primary' | 'secondary' | 'error'
+  variant?: 'bordered' | 'contained' | 'outlined'
+}
+
+interface ButtonsProps {
+  cancelButtonProps?: ButtonProps
+  submitButtonProps?: ButtonProps
```

Comment:
perhaps `confirmButton` it's more generic?

Reply:
ahh, thanks!